<?xml version="1.0"?>

<launch>
    <rosparam command="load" file="$(find gazebo_drl_env)/param/forceFactors.yaml" />   
    <rosparam command="load" file="$(find gazebo_drl_env)/param/robot_start.yaml" />   
    <param name="TB3_AS_ACTOR" type="bool" value="true" /> 
    <param name="BASE_WORLD" type="string" value="corridor4m.world" /> 
   
    <param name="ACTOR_NUMBER" type="int" value="1" />
    <param name="ROBOT_X_START" value="-5.0" />
    <param name="ROBOT_X_END" value="-5.1" />
    <param name="ROBOT_Y_START" value="0.0" />
    <param name="ROBOT_Y_END" value="0.0" />
    <param name="ROBOT_YAW_START" value="0.0" />
    <param name="ROBOT_YAW_END" value="0.1" />
    <param name="TARGET_X"  value="5.0" />
    <param name="TARGET_Y"  value="0.0" />
    
    <!-- creating world and define several tf relationship -->
    <node name="create_world" pkg="actor_services" type="overtaking_wall_1.py"/>
    <node name="tfodom2defaultworld" pkg="turtlebot3_social" type="tfodom2defaultworld.py"/>
    <node name="tfped2defaultworld" pkg="turtlebot3_social" type="tfped2defaultworld.py"/>

    <arg name="world_file"  default="$(find gazebo_drl_env)/worlds/ped_world.world"/>
    <arg name="gui" default="true"/>
    <arg name="rviz" default="false"/>
    <arg name="use_sim_time" value="true"/>

    <!-- set use_sim_time flag -->
    <group if="$(arg use_sim_time)">
        <param name="/use_sim_time" value="true" />
    </group>

    <!-- start rviz -->
    <group if="$(arg rviz)">
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find gazebo_drl_env)/rviz/tf_show.rviz"/>
    </group>
    <!-- start gazebo client -->
    <node name="gazebo" pkg="gazebo_ros" type="gzserver" output="screen" args="$(arg world_file)" />
    <group if="$(arg gui)">
        <node name="gazebo_gui" pkg="gazebo_ros" type="gzclient" respawn="false" output="screen"/>
    </group>

    <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle]"/>
    <arg name="x_pos" default="0.0"/>
    <arg name="y_pos" default="0.0"/>
    <arg name="z_pos" default="0.0"/>

    <param name="robot_description" command="$(find xacro)/xacro.py $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model turtlebot3_burger -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -param robot_description" />

    <!--<include file="$(find turtlebot3_teleop)/launch/turtlebot3_teleop_key.launch" />-->
    
    <!--<node name="main_service" pkg="Gazebo_DRL_env" type="task_env_service" launch-prefix="xterm -e gdb -\-args"/>-->
    <node name="main_service" pkg="gazebo_drl_env" type="task_env_service" />
    <node name="target2worldbr" pkg="gazebo_drl_env" type="target2worldbr.py"/>
</launch>
